src/System.Buffers.Primitives/System/Buffers/Buffer.cs

```diff
@@ -26,11 +26,48 @@ internal Buffer(OwnedBuffer<T> owner, int index, int length)
             _length = length;
         }
 
-        internal Buffer(T[] array, int index, int length)
+        public Buffer(T[] array)
```

Comment:
I can't do this since array could be null and then array.Length would throw null exception.